src/writer/tests.rs

```diff
@@ -18,7 +18,9 @@ proptest! {
             writer.flush().unwrap();
         }
         let mut buf = vec![0; bytes.len()];
-        memory.read(offset, &mut buf);
+        if !bytes.is_empty(){
```

Comment:
@ielashi After adding the file to proptest-regressions it revealed that `should_write_single_slice` is flaky for the same input. Hence to pass CI both tests should be fixed in a single PR.

Reply:
@dragoljub-duric Thanks for noticing. Looking.

Reply:
Rather than adding this exception, I'd suggest we change the proptest to not include bytes of zero length.

nit: let's also update the PR description/title accordingly.